### PR TITLE
Thorium Brotherhood

### LINKED ISSUE #1074: The Thorium Brotherhood
**Describe your suggestion in full detail below:**
 After a "non-evil" faith Kingdom-tier ruler gets the control of the Burning Steppes or Chiselgrip, nearby rulers in Khaz Modan and Azeroth should have a decision like "Invite the Thorium Brotherhood" where a highly talented inspired dwarf would visit the rulers Royal Court to craft an artifact and also get a county modifier at their capital named "Invited Talented Craftsmen" with effects like decreased build cost and time, increased development gain and +5% Taxes. If the player doesn't possess the Royal Court DLC, they would only get the county modifier.

### PR DESCRIPTION
**Pull request by Malicos**
## Changelog:
- Added a decision to invite the Thorium Brotherhood craftsmen if Chiselgrip or Drognur are held by non-Dark Plural character.

## Developer changelog:
- Added a decision.
- Added modifiers related to the TB craftsmen and bases.
- Added an on_action regulating the appearance of base modifiers.
- Added an event related to the aforementioned decision allowing to pick an inspiration for the invited craftsman.
- Added a character template for the spawnable craftsmen.
- Added a scripted trigger checking whether holders of Drognur or Chiselgrip are Dark Plural, used in decision.

## TODO:
- [x] Check whether ruler is in Azeroth or Khaz Modan regions
- [x] Add province modifiers if Chiselgrip or Drognur (Thorium Point) are held by non-Dark Plural character.
- [x] Remove province modifiers if any of the two are held by Dark Plural faith characters.
- [x] Make the event spawn an inspired courtier and province modifier for the person picking the decision.
- [x] Event localization

## Tests:
- [ ] There are no errors in `wc` files in `Documents\Paradox Interactive\Crusader Kings III\logs\error.log` except `portrait_decals.cpp:101`
- [ ] The mod takes less than 5.5 GB in the Task Manager (Windows)

# How to test:
- To check whether Thorium Brotherhood base modifiers appear, play as holder of either Chiselgrip or Drognur and switch their religion to non-Dark Plural. The modifier should appear on yearly pulse. To make them disappear, simply change religion back to Dark Plural.
- To invite Thorium Brotherhood craftsmen, either Chiselgrip or Drognur need to be held by a character with non-Dark Plural faith. Simply change the faith of the characters in question in console and switch to any character in de jure Azeroth or Khaz Modan empires. Give them necessary things like gold and piety in order to activate the decision. It should spawn a courtier and give a county modifier in your capital province for 10 years.

### CODE REVIEW

File: common/on_action/wc_thorium_on_actions.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	effect = {
+		trigger_event = { on_action = set_thorium_base_modifier }
+	}
+}
+	
+set_thorium_base_modifier = {
+	effect = {
+		set_thorium_base_effect = yes
+	}
+}
```
Comment (robmart) [inline common/on_action/wc_thorium_on_actions.txt]: These two can be condensed into one

**Review by robmart — CHANGES_REQUESTED** [review]

File: common/decisions/wc_thorium_decisions.txt [hunk 2]
```diff
@@ -0,0 +1,63 @@
+﻿# Warcraft
```
Comment (robmart) [inline common/decisions/wc_thorium_decisions.txt]: You don't need to have `# Warcraft` in files starting with `wc_`, only in files that are from vanilla

**Review by robmart — APPROVED** [review]
LGTM

**Review by robmart — CHANGES_REQUESTED** [review]
loc

File: localization/english/event_localization/wc_thorium_brotherhood_l_english.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿l_english:
+ thorium_brotherhood.1.t:0 "Choosing a Craft"
+ thorium_brotherhood.1.desc:0 "Placeholder"
+ thorium_brotherhood.1.a:0 "A weapon."
+ thorium_brotherhood.1.b:0 "An armor."
+ thorium_brotherhood.1.c:0 "A metalsmith."
```
Comment (edmund-h) [inline localization/english/event_localization/wc_thorium_brotherhood_l_english.yml]:
```suggestion
 thorium_brotherhood.1.t:0 "The Inspiration Globe"
 thorium_brotherhood.1.desc:0 "The descent to the castle workshop is one I do not make often. The air is hot and stuffy, and I find myself sweating in a way most unbecoming of a [ROOT.Char.GetTitleAsNameNoTooltip]. Still, it is a duty I must perform myself to show our esteemed guest of the Thorium Brotherhood the utmost hospitality.\n\nUpon inspecting the forge and other facilities and finding them to be adequate, [craftsman.Char.GetName] places his pack on the worktable and addresses me.\n\n\"[ROOT.Char.GetTitledFirstName], it is time for you to make a choice.\"As [craftsman.Char.GetSheHe] speaks, [craftsman.Char.GetFirstName] produces a softly glowing, transparent orb from [craftsman.Char.GetHerHis] pack. \"I present to yo the Orb of Inspiration, an heirloom passed down through generations of Thorium Brothers. Its functionality and purpose are quite intuitive. Simply place your hand on the front of the orb and imagine an item you would like me to create for you. Then the image of that item will appear in my own mind along with the best possible method of creating it.\"\n\nI have never heard of such magic before, but in the end it does make sense to ease communication between craftsman and client. With little more than a moment's hesitation, I place my hand on the orb and think about..."
 thorium_brotherhood.1.a:0 "...a weapon."
 thorium_brotherhood.1.b:0 "...a suit of armor."
 thorium_brotherhood.1.c:0 "...an elegant trinket."
```

File: common/decisions/wc_thorium_decisions.txt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		OR = {
+			any_held_title = { # Must be in Azeroth region
+				tier = tier_county
+				title_province = { geographical_region = world_eastern_kingdoms_azeroth }
+			}
+			any_held_title = { # Must be in Khaz Modan region
+				tier = tier_county
+				title_province = { geographical_region = world_eastern_kingdoms_khaz_modan }
+			}
+		}
```
Comment (robmart) [inline common/decisions/wc_thorium_decisions.txt]:
```suggestion
	any_held_title = { # Must be in Khaz Modan region
		tier = tier_county
		OR = {
			title_province = { geographical_region = world_eastern_kingdoms_khaz_modan }
			title_province = { geographical_region = world_eastern_kingdoms_azeroth }
		}
	}
```

Reduces lag

File: common/modifiers/wc_event_modifiers.txt [hunk 5]
```diff
@@ -313,4 +313,22 @@ wc_thunder_king_ai_modifier = { # AI Thunder King only
...
+wc_thorium_craftsmen_modifier = {
+	icon = county_modifier_development_positive
+	development_growth_factor = 0.1
+	tax_mult = 0.05
+	build_gold_cost = -0.15
+	build_speed = -0.15
+}
+
+wc_thorium_base_modifier = {
+	icon = martial_positive
```
Comment (robmart) [inline common/modifiers/wc_event_modifiers.txt]: Martial icon?

**Review by robmart — CHANGES_REQUESTED** [review]
Minor changes

File: common/modifiers/wc_event_modifiers.txt (hunk 5 above)
Comment (Malicos) [reply common/modifiers/wc_event_modifiers.txt]: it adds mostly martial bonuses, can change if necessary

Comment (Malicos) [reply common/modifiers/wc_event_modifiers.txt]: changed

**Review by robmart — APPROVED** [review]
LGTM

**Pull request merged**